src/Middleware/ThrottleMiddleware.php

```diff
@@ -0,0 +1,221 @@
+<?php
+namespace Muffin\Throttle\Middleware;
+
+use Cake\Cache\Cache;
+use Cake\Core\InstanceConfigTrait;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Diactoros\Stream;
+
+class ThrottleMiddleware
+{
+
+    use InstanceConfigTrait;
+
+
+    /*
+     * Default config for Throttle Middleware
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'message' => 'Rate limit exceeded',
+        'interval' => '+1 minute',
+        'limit' => 10,
+        'headers' => [
+            'limit' => 'X-RateLimit-Limit',
+            'remaining' => 'X-RateLimit-Remaining',
+            'reset' => 'X-RateLimit-Reset'
+        ]
+    ];
+
+    /**
+     * Cache configuration name
+     *
+     * @var string
+     */
+    public static $cacheConfig = 'throttle';
+
+    /**
+     * Cache expiration key suffix
+     *
+     * @var string
+     */
+    public static $cacheExpirationSuffix = 'expires';
+
+    /**
+     * Unique client identifier
+     *
+     * @var string
+     */
+    protected $_identifier;
+
+    /**
+     * Number of connections after increment
+     *
+     * @var string
+     */
+    protected $_count;
+
+    /**
+     * ThrottleMiddleware constructor.
+     *
+     * @param array $config Configuration options
+     */
+    public function __construct($config = [])
+    {
+        $config += [
+            'identifier' => function (ServerRequestInterface $request) {
+                return $request->clientIp();
+            }
+        ];
+        $this->setConfig($config);
+    }
+
+
+    /**
+     * Called when the class is used as a function
+     *
+     * @param ServerRequestInterface $request Request object
```

Comment:
Please use full namespace like `\Psr\Http\Message\ServerRequestInterface` starting with backslash for all type declarations in docblocks.